haskell/toolchain.bzl

```diff
@@ -56,6 +71,49 @@ def _run_ghc(hs, cc, inputs, outputs, mnemonic, arguments, env, params_file = No
     hs.actions.write(compile_flags_file, args)
     hs.actions.write(extra_args_file, arguments)
 
+    if abi_file != None:
+        # We declare the 2 files containing informations about the call to GHC with --show-iface.
+        compiler_file = hs.actions.declare_file("compiler_%s" % (hs.name))
+        flags_file = hs.actions.declare_file("flags_%s_%s" % (hs.name, extra_name))
+
+        # The first one contains the path to GHC
+        ghc_args = hs.actions.args()
+        ghc_args.set_param_file_format("multiline")
+        ghc_args.add(hs.tools.ghc.path)
+        hs.actions.write(compiler_file, ghc_args)
+
+        # The second one contains the "--show-iface" option and the name of the hi file.
+        flag_args = hs.actions.args()
+        flag_args.set_param_file_format("multiline")
+        flag_args.add("--show-iface ")
+        flag_args.add(hi_file.path)
+        hs.actions.write(flags_file, flag_args)
```

Comment:
The `ghc_wrapper` could produce these flags itself when extracting the ABI is required. It doesn't seem necessary to assemble them in the caller side.